packages/sw-background-sync-queue/src/lib/queue-utils.js

```diff
@@ -0,0 +1,94 @@
+import {getDb} from './background-sync-idb-helper';
+
+/**
+ * takes a request and gives back JSON object that is storable in IDB
+ *
+ * @param {Request} request request object to transform
+ * into iDB storable object
+ * @param {Object} config config object to be
+ * stored along in the iDB
+ * @return {Object} indexable object for iDB
+ *
+ * @memberOf RequestManager
+ */
+async function getQueueableRequest(request, config) {
+	let requestObject={
+		config: config,
+		metadata: {
+			creationTimestamp: Date.now(),
+		},
+	};
+	let headerObject = {};
+	request.headers.forEach((value, field)=> {
+		headerObject[field] = value;
+	});
+	requestObject.request = {
+		url: request.url,
+		headers: headerObject,
+		mode: request.mode,
+		method: request.method,
+		redirect: request.redirect,
+	};
+	let requestBody = await request.text();
```

Comment:
I think you still have the issue of this failing when the request body contains non-UTF16 data?